src/plots/cartesian/axis_defaults.js

```diff
@@ -64,6 +65,10 @@ module.exports = function handleAxisDefaults(containerIn, containerOut, coerce,
         }
     }
 
+    containerOut._initialCategories = axType === 'category' ?
+        orderedCategories(letter, containerIn.categorymode, containerIn.categorylist, options.data) :
```

Comment:
and @monfera I should add, patterns like :arrow_double_up: are common in plotly.js. 

Don't try to reinvent the wheel. As mentioned before, this [block](https://github.com/plotly/plotly.js/blob/fb874424d5b89c021ef6e5ebe0326433389d02ee/src/plots/cartesian/tick_value_defaults.js#L23-L27) should serve as a good guide for how `categorymode` and `categorylist` should interact.